scalyr_agent/configuration.py

```diff
@@ -367,12 +372,118 @@ def parse(self):
 
             self.__monitor_configs = list(self.__config.get_json_array("monitors"))
 
-            self.__api_keys_configs = list(self.__config.get_json_array("api_keys"))
+            # NOTE do this verifications only after all config fragments were added into configurations.
+            self.__verify_api_keys()
+            self.__verify_and_match_api_keys_in_logs()
+
+            self.__api_key_configs = list(self.__config.get_json_array("api_keys"))
 
         except BadConfiguration as e:
             self.__last_error = e
             raise e
 
+    def __verify_api_keys(self):
+        """
+        Verify all api key config entries from the "api_keys" list in config.
+        """
+        api_keys = self.__config.get_json_array("api_keys")
+        default_key_config = None
+        for key_config in api_keys:
+            api_key = key_config.get("api_key", none_if_missing=True)
+            # if the 'api_key' from this entry equals to the 'main' api key,
+            # than this api key entry is considered as default.
+            if api_key is None or api_key == self.api_key:
+                if default_key_config is None:
+                    default_key_config = key_config
+                else:
+                    # if there are more than one api key entries that also can be considered as a default,
+                    # then raise a duplication error
+                    raise BadConfiguration(
+                        "The default api key is redefined. Please use only unique api keys."
+                        "The entries without 'api_key' field are also considered as default",
+                        "api_keys",
+                        "apiKeyDuplication",
+                    )
+
+        unique_keys = set()
+        other_api_key_configs = list()
+        # If there are multiple api key entries with the same api_key, then the duplication error is raised too.
+        for key_config in api_keys:
```

Comment:
As discussed, change this to exclude entries with the same `id` and not `api_key` value.